protocol/lavasession/single_provider_session.go

```diff
@@ -170,6 +194,17 @@ func (sps *SingleProviderSession) PrepareSessionForUsage(ctx context.Context, cu
 		sps.lock.Unlock() // unlock on error
 		return err
 	}
+
+	// Update badgeUsedCu in ProviderSessionsWithConsumer
+	if badgeUser != "" {
```

Comment:
if badgeSession != nil {

Reply:
Also fixed in the same commit above: 704563ba77bb3d13a013adcc88dce17aba97627a